cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -565,3 +565,19 @@ private class MyConsistencyConfiguration extends Consistency::ConsistencyConfigu
     any()
   }
 }
+
+/**
+ * Gets an additional term that is added to the `join` and `branch` computations to reflect
+ * an additional forward or backwards branching factor that is not taken into account
+ * when calculating the (virtual) dispatch cost.
+ *
+ * `call` is a call with an argument `arg` that is part of a path from a source to a sink, and
+ * `p` is the target parameter of a callable to which `call` may resolve.
+ *
+ * All these values are bound by the dataflow library, and if this predicate is implemented it
+ * should be specified with a bindingset annotation that binds all the columns.
+ */
+bindingset[call, p, arg]
+int getAdditionalFlowIntoCallNodeTerm(DataFlowCall call, ParameterNode p, ArgumentNode arg) {
```

Comment:
Let's remove `call`, as it is superfluous (it is given by `arg`). Also, let's drop the bindingset - ideally we'll want this input predicate nomagic'ed in the cases where it's non-trivial.

Reply:
Also, I'd prefer to switch the order of the `arg` and `p` columns.

Reply:
Fixed in 93b965302b0d20626e6864840b3401aedd2f63cb and ce96973e5033450ffd419b26be456aa76cbc3404.